### PR TITLE
Chap9 1

### PR DESCRIPTION
**Pull request by nguyenhop14**

### CODE REVIEW

**Comment by nguyenhop14** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -3,16 +3,36 @@ def log_in user
...
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
-    return unless session[:user_id]
-    @current_user ||= User.find_by(id: session[:user_id])
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by id: user_id
+    elsif (user_id = cookies.signed[:user_id])
```
Comment (huuphuc93) [inline app/helpers/sessions_helper.rb]: xóa ()

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -3,16 +3,36 @@ def log_in user
...
+      user = User.find_by id: user_id
+      if user && user.authenticated?(cookies[:remember_token])
```
Comment (huuphuc93) [inline app/helpers/sessions_helper.rb]:
```suggestion
      if user&.authenticated?(cookies[:remember_token])
```

File: app/models/user.rb [hunk 3]
```diff
@@ -14,5 +15,23 @@ def digest string
...
+    update remember_digest: User.digest(remember_token)
+  end
+
+  def authenticated?(remember_token)
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+  def forget
+    update_attribute(:remember_digest, nil)
```
Comment (huuphuc93) [inline app/models/user.rb]: update thay cho update_attribute

**Comment by nguyenhop14** [comment]
ready

**Comment by huuphuc93** [comment]
commented
gộp commits rồi merged nhé em

**Pull request merged**